prereise/gather/hiflddata/plant_agg.py

```diff
@@ -0,0 +1,211 @@
+import csv
+
+import pandas as pd
+from geopy.distance import geodesic
+
+from prereise.gather.hiflddata.data_trans import Clean, get_Zone
+
+Max_Value = 3000
+Min_Value = 0
+coord_precision = ".9f"
+
+
+def Clean_p(P_csv):
+    csv_data = pd.read_csv(P_csv)
+    Num_sub = len(csv_data)
+    row_indexs = []
+    for i in range(Num_sub):
+        if csv_data["STATUS"][i] != "OP":
+            row_indexs.append(i)
+    clean_data = csv_data.drop(labels=row_indexs)
+    return clean_data
+
+
+def LocOfsub(clean_data):
+    """Get the latitude and longitude of substations, and the substations in the area of each zip code
+
+    :param dict clean_data:  a dict of substations as returned by :func:`data_trans.Clean`
+    :return: (*dict*) -- LocOfsub_dict, dict mapping the geo coordinate (x,y) to substations.
+    :return: (*dict*) -- ZipOfsub_dict, dict mapping the zip code to a group of substations.
+    """
+
+    LocOfsub_dict = {}
+    ZipOfsub_dict = {}
+    for index, row in clean_data.iterrows():
+        loc = (
+            format(row["LATITUDE"], coord_precision),
+            format(row["LONGITUDE"], coord_precision),
+        )
+        sub = row["ID"]
+        zi = row["ZIP"]
+        LocOfsub_dict[sub] = loc
+        if zi in ZipOfsub_dict:
+            list1 = ZipOfsub_dict[zi]
+            list1.append(sub)
+            ZipOfsub_dict[zi] = list1
+        else:
+            list1 = [sub]
+            ZipOfsub_dict[zi] = list1
+    return LocOfsub_dict, ZipOfsub_dict
+
+
+def Cal_P(G_csv):
+    """Calculate the Pmin for each plant
+
+    :param dict G_csv: path of the EIA generator profile csv file
+    :return: (*list*) -- a list of Pmin for the plants.
+    """
+
+    Pmin = {}
+    csv_data = pd.read_csv(G_csv)
+    for index, row in csv_data.iterrows():
+        Pmin[str(row["Plant Name"]).upper()] = row["Minimum Load (MW)"]
+    return Pmin
+
+
+def Loc_of_plant():
+    """Get the latitude and longitude of plants
+
+    :return: (*list*) -- a list of power plants' geo coordinates.
+    """
+
+    loc_of_plant = {}
+    csv_data = pd.read_csv("data/Power_Plants.csv")
+    for index, row in csv_data.iterrows():
+        loc = (
+            format(row["LATITUDE"], coord_precision),
+            format(row["LONGITUDE"], coord_precision),
+        )
+        loc_of_plant[row["NAME"]] = loc
+    return loc_of_plant
+
+
+def Plant_agg(clean_data, ZipOfsub_dict, loc_of_plant, LocOfsub_dict, Pmin):
+    """Aggregate the plant by zip code and build the plant dict with geo-based aggregation
+
+    :return: (*dict*) -- a dict for power plant after aggregation
```

Comment:
Can you be more specific with what exactly the keys and values of this dict represent? It seems that each value is an unlabeled list of floats, which makes it hard to understand what's going on in the `write_plant` function which is heavily dependent on the position of these values.

Reply:
Will add details in next revision.